src/compile/model.ts

```diff
@@ -64,6 +70,11 @@ class NameMap {
   }
 }
 
+export interface NameMapShape {
```

Comment:
Can you explain why you need to add this interface?  

In any case, `NameMapShape` is quite unconventional IMHO.  Why not `NameMapInterface`?